pkg/proxy/oxyproxy.go

```diff
@@ -344,83 +184,61 @@ func createOxyRedirect(l4 *policy.L4Filter, id string, source ProxySource, to ui
 		source:  source,
 		router:  route.New(),
 		ingress: l4.Ingress,
-		nodeInfo: accesslog.NodeAddressInfo{
-			IPv4: node.GetExternalIPv4().String(),
-			IPv6: node.GetIPv6().String(),
-		},
 	}
 
 	redir.epID = source.GetID()
 
 	redirect := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		record := &accesslog.LogRecord{
-			Request:           req,
-			Timestamp:         time.Now().UTC().Format(time.RFC3339Nano),
-			NodeAddressInfo:   redir.nodeInfo,
-			TransportProtocol: 6, // TCP's IANA-assigned protocol number
-		}
-
-		if redir.ingress {
-			record.ObservationPoint = accesslog.Ingress
-		} else {
-			record.ObservationPoint = accesslog.Egress
-		}
+		record := newHTTPLogRecord(redir, req.Method, req.URL, req.Proto, req.Header)
 
-		srcIdentity, dstIPPort, err := lookupNewDestFromHttp(req, to)
+		srcIdentity, dstIPPort, err := lookupNewDest(req.RemoteAddr, to)
 		if err != nil {
 			// FIXME: What do we do here long term?
 			log.Errorf("%s", err)
 			http.Error(w, err.Error(), http.StatusBadRequest)
-			record.Info = fmt.Sprintf("cannot generate url: %s", err)
-			accesslog.Log(record, accesslog.TypeRequest, accesslog.VerdictError, http.StatusBadRequest)
+			record.log(accesslog.TypeRequest, accesslog.VerdictError,
+				http.StatusBadRequest, fmt.Sprintf("cannot generate url: %s", err))
 			return
 		}
 
-		info, version := redir.getSourceInfo(req, policy.NumericIdentity(srcIdentity))
-		record.SourceEndpoint = info
-		record.IPVersion = version
-
-		if srcIdentity != 0 {
-			record.SourceEndpoint.Identity = uint64(srcIdentity)
-		}
-
-		record.DestinationEndpoint = redir.getDestinationInfo(dstIPPort)
+		record.fillInfo(redir, req.RemoteAddr, dstIPPort, srcIdentity)
 
+		var info string
 		// Validate access to L4/L7 resource
 		redir.mutex.Lock()
 		if len(redir.rules) > 0 {
 			rule, _ := redir.router.Route(req)
 			if rule == nil {
 				http.Error(w, "Access denied", http.StatusForbidden)
 				redir.mutex.Unlock()
-				accesslog.Log(record, accesslog.TypeRequest, accesslog.VerdictDenied, http.StatusForbidden)
+				record.log(accesslog.TypeRequest, accesslog.VerdictDenied,
+					http.StatusForbidden, "")
 				return
 			}
 			ar := rule.(string)
-			log.Debugf("Allowing request based on rule %+v", ar)
-			record.Info = fmt.Sprintf("rule: %+v", ar)
+			log.WithField(logfields.Object, fmt.Sprintf("%+v", "Allowing request based on rule"))
```

Comment:
This is missing the Debug call
```golang
log.WithField(logfields.Object, fmt.Sprintf("%+v", ar).Debug("Allowing request based on rule")
```
We can also backport the logfields.Repr function but it basically just does the fmt.Sprintf call

Reply:
fixed